provider/pihole/clientV6.go

```diff
@@ -0,0 +1,376 @@
+	if endpoint.RecordTypeCNAME == rtype {
+		results = apiResponse.Config.DNS.CnameRecords
+	} else {
+		results = apiResponse.Config.DNS.Hosts
+	}
+
+	return results, nil
+}
+
+func (p *piholeClientV6) listRecords(ctx context.Context, rtype string) ([]*endpoint.Endpoint, error) {
+	out := make([]*endpoint.Endpoint, 0)
+	results, err := p.getConfigValue(ctx, rtype)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rec := range results {
+		recs := strings.FieldsFunc(rec, func(r rune) bool {
+			return r == ' ' || r == ','
+		})
+		var DNSName string
+		var Target string
+		var Ttl endpoint.TTL
+		// A/AAAA record format is target(IP) DNSName
+		DNSName = recs[1]
+		Target = recs[0]
+		switch rtype {
+		case endpoint.RecordTypeA:
+			if strings.Contains(recs[0], ":") {
+				continue
+			}
+		case endpoint.RecordTypeAAAA:
+			if strings.Contains(recs[0], ".") {
+				continue
+			}
+		case endpoint.RecordTypeCNAME:
+			// CNAME format is DNSName,target
+			DNSName = recs[0]
+			Target = recs[1]
+			if len(recs) == 3 { // TTL is present
+				// Parse string to int64 first
+				ttlInt, err := strconv.ParseInt(recs[2], 10, 64)
+				if err != nil {
+					// Handle parsing error or fail test
+					log.Warnf("failed to parse TTL value '%s': %v; using a TTL of 0", recs[2], err)
+				}
+				Ttl = endpoint.TTL(ttlInt)
+			}
+		}
+
+		out = append(out, &endpoint.Endpoint{
+			DNSName:    DNSName,
+			Targets:    []string{Target},
+			RecordTTL:  Ttl,
+			RecordType: rtype,
+		})
+	}
+	return out, nil
+}
+
+func (p *piholeClientV6) createRecord(ctx context.Context, ep *endpoint.Endpoint) error {
+	return p.apply(ctx, http.MethodPut, ep)
+}
+
+func (p *piholeClientV6) deleteRecord(ctx context.Context, ep *endpoint.Endpoint) error {
+	return p.apply(ctx, http.MethodDelete, ep)
+}
+
+func (p *piholeClientV6) aRecordsScript() string {
+	return fmt.Sprintf("%s/api/config/dns/hosts", p.cfg.Server)
+}
+
+func (p *piholeClientV6) cnameRecordsScript() string {
+	return fmt.Sprintf("%s/api/config/dns/cnameRecords", p.cfg.Server)
+}
+
+func (p *piholeClientV6) urlForRecordType(rtype string) (string, error) {
+	switch rtype {
+	case endpoint.RecordTypeA, endpoint.RecordTypeAAAA:
+		return p.aRecordsScript(), nil
+	case endpoint.RecordTypeCNAME:
+		return p.cnameRecordsScript(), nil
+	default:
+		return "", fmt.Errorf("unsupported record type: %s", rtype)
+	}
+}
+
+// ApiAuthResponse Define a struct to match the JSON response /auth/app structure
+type ApiAuthResponse struct {
+	Session struct {
+		Valid    bool   `json:"valid"`
+		TOTP     bool   `json:"totp"`
+		SID      string `json:"sid"`
+		CSRF     string `json:"csrf"`
+		Validity int    `json:"validity"`
+		Message  string `json:"message"`
+	} `json:"session"`
+	Took float64 `json:"took"`
+}
+
+// ApiErrorResponse Define struct to match the JSON structure
+type ApiErrorResponse struct {
+	Error struct {
+		Key     string `json:"key"`
+		Message string `json:"message"`
+		Hint    string ` json:"hint"`
```

Comment:
Is this supposed to have that blank?
Or maybe this is correct?
```suggestion
		Hint    string `json:"hint"`
```